next-data/blogData.ts

```diff
@@ -0,0 +1,19 @@
+import { ENABLE_STATIC_EXPORT, NEXT_DATA_URL } from '@/next.constants.mjs';
+import type { BlogDataRSC } from '@/types';
+
+const getBlogData = (category: string): Promise<BlogDataRSC> => {
+  if (ENABLE_STATIC_EXPORT) {
+    // Loads the data dynamically with lazy-loading to prevent data-generation
+    // within the top level import
+    return import('@/next-data/providers/blogData').then(
```

Comment:
there's perhaps a reason here i am not thinking of but this file notably is not using async/await